src/bundle/Controller/UserSettingsController.php

```diff
@@ -104,12 +104,24 @@ public function updateAction(Request $request, UpdateView $view)
                     'ibexa_user_settings'
                 );
 
+                $returnUrl = $request->query->has('returnUrl') ? $request->query->get('returnUrl') : null;
+
                 if ($form->getClickedButton() instanceof Button
                     && $form->getClickedButton()->getName() === UserSettingUpdateType::BTN_UPDATE_AND_EDIT
                 ) {
-                    return $this->redirectToRoute('ibexa.user_settings.update', [
+                    $routeParams = [
                         'identifier' => $data->getIdentifier(),
-                    ]);
+                    ];
+                    
+                    if ($returnUrl) {
+                        $routeParams['returnUrl'] = $returnUrl;
+                    }
+
+                    return $this->redirectToRoute('ibexa.user_settings.update', $routeParams); 
+                }
+
+                if ($returnUrl) {
+                    return $this->redirect($returnUrl);
```

Comment:
So if I create a request like `?returnUrl=https%3A%2F%2Fmy-mailicious-site.invalid%2F` will that result in a redirect to an external site?